global/scripts/RSNregression.m

```diff
@@ -178,6 +178,56 @@ function RSNregression(InputFile, InputVNFile, Method, ParamsFile, OutputBeta, v
             
             betaICAone = weightedDualRegression(GroupMapcii.cdata, inputConcat, AreaWeights .* ScaledAlignmentQuality);
             [betaICA, NODEts] = weightedDualRegression(normalise(betaICAone), inputConcat, AreaWeights);
+            
+            if strcmp(Method, "tICA_weighted") % based on tICA/scripts/ComputeGroupTICA.m but for one subject as a group
+                NODEts=NODEts';
+                disp(['NODEts'])
+                size(NODEts)
+                SpectraArray = textscan(optional.SpectraParams, '%s', 'Delimiter', {'@'});
+                nTPsForSpectra =str2double(SpectraArray{1}{1});
+                thisStart = 1;
+                TCSRunVarSub = [];
+                for i = 1:size(inputArray, 1)
+                    dtseriesName=[inputArray{i}];
+                    if exist(dtseriesName, 'file')
+                        [~, runLengthStr] = system(['wb_command -file-information -only-number-of-maps ' dtseriesName]);
+                        disp(['runLengthStr ' runLengthStr])
+                        runLength = str2double(runLengthStr);
+                        nextStart = thisStart + runLength;
+                        TCSRunVarSub = [TCSRunVarSub repmat(std(NODEts(:, thisStart:(nextStart - 1)), [], 2), 1, runLength)];
+    
+                        thisStart = nextStart;
+                    end
+                end
+                
+                sICAtcsvars = std(NODEts, [], 2);
+                NODEts = (NODEts ./ TCSRunVarSub) .* repmat(sICAtcsvars, 1, size(TCSRunVarSub, 2)); %Making all runs contribute equally improves tICA decompositions
+                NODEts(~isfinite(NODEts)) = 0;
+
+                A = load(optional.tICAMM);
+
+                if size(A,1) ~= size(NODEts,1)
+                    error('Mixing matrix to be used does not match dimensions of the sICA components');
+                end
+                
+                % unmix the sICA timeseries
+                W = pinv(A);
+                normicasig = W * NODEts;
+                icasig = normicasig .* repmat(std(A ./ repmat(sICAtcsvars, 1, size(A, 2)))', 1, size(NODEts, 2)); %Unormalize the icasig assuming sICAtcs with std = 1 (approximately undo the original variance normalization)
+                
+                tICAtcs = icasig';
+                TSTDs = std(tICAtcs, [], 1); %unnormalized tICA temporal standard deviations
+                TIs = [1:1:length(TSTDs)];
+
+                tICAtcs(:, [1:1:length(TIs)]) = single(tICAtcs(:, TIs));
+                tICAtcs=tICAtcs';
+                tICAtcsAll = reshape(tICAtcs, size(A,1), nTPsForSpectra, 1);
+                tICATCS = squeeze(tICAtcsAll(:, std(tICAtcsAll(:, :, 1), [], 1) > 0, 1));
```

Comment:
`range(...) > 0` is much more stable (and a bit faster) than `std(...) > 0`.

Reply:
Also, if this is testing for timepoints that are all zeros, I'm not sure that is even necessary in this code (we don't have the pad-subjects-to-expected-timepoints stuff in play here, right?).  With the several transposes in the code, I'm not sure exactly what dimension is what, though.